pkg/component/component.go

```diff
@@ -929,11 +923,12 @@ func ListS2IComponents(client *occlient.Client, applicationName string, localCon
 
 		// create a list of object metadata based on the component and application name (extracted from DeploymentConfig labels)
 		for _, elem := range dcList {
-			component, err := GetComponent(client, elem.Labels[componentlabels.ComponentLabel], applicationName, client.Namespace)
+			component, err := GetComponent(client, elem.Labels[componentlabels.ComponentLabel], elem.Labels[applabels.ApplicationLabel], client.Namespace)
 			if err != nil {
 				return ComponentList{}, errors.Wrap(err, "Unable to get component")
 			}
```

Comment:
I have a question here. Why do we return if `err!=nil` even though there are some other checks pending in the function, for e.g. this check - https://github.com/openshift/odo/pull/4742/files#diff-a84453ec52dba47b60234fd3aba56849e52e2bae2577c082dcf1a79ef6b25855R946 ?
```
if client != nil {
```

Why do we avoid that and return? Shouldn't we check for `if err == nil { continue }` and execute rest of the code? I have seen this type of logic in almost all places. For some it makes sense, but for others I think we should continue executing the program and if it does not make sense to do so, then perhaps we should add a comment about why we are returning without executing the rest of the code.

Other examples are - 
1. pkg/odo/cli/component/list.go:264 - https://github.com/openshift/odo/pull/4742/files#diff-a5ee7fb0c76cd592580dce46cda117d8a483452979b0da40d644906f88cba343R264
2. pkg/odo/cli/component/list.go:274 - https://github.com/openshift/odo/pull/4742/files#diff-a5ee7fb0c76cd592580dce46cda117d8a483452979b0da40d644906f88cba343R274
3. ..and many more